xbmc/addons/AddonEvents.h

```diff
@@ -48,6 +48,28 @@ namespace ADDON
       MetadataChanged(std::string id) : id(std::move(id)) {}
     };
 
+    /**
+     * Emitted when a different version of the add-on has been installed
+     * to the file system and should be reloaded.
+     */
+    struct ReInstalled: AddonEvent
+    {
+      std::string id;
+      ReInstalled(std::string id) : id(std::move(id)) {}
+    };
+
+    /**
+     * Emitted after the add-on has been uninstalled.
+     */
+    struct UnInstalled : AddonEvent
+    {
+      std::string id;
+      UnInstalled(std::string id) : id(std::move(id)) {}
```

Comment:
It doesn't really matter if something is stopped a few moments after uninstall instead of before, that's why.
Services have actually always been stopped ascync on uninstall even though Stop was called in the pre-hook (since CScriptInvocationManager::Stop is non-blocking).